### ISSUE TITLE
[Usage question] - how to access the public IBL database after setting up the internal database

### ISSUE THREAD

**Issue by bindeel** [int-brain-lab/iblenv#382]
When attempting to troubleshoot some missing properties in my ONE set-up, I changed silent=True to silent=False in:

from one.api import ONE
ONE.setup(base_url='https://openalyx.internationalbrainlab.org', silent=True)
one = ONE(password='international')

I now realize that this changed my settings to connect to the internal IBL database, as it asked me to type my Alyx credentials. However, I don't have an Alyx account, so I now get this error when I run the ONE setup again:

HTTPError: Failed to load the remote cache file

I get this error even if I change silent=False back to the original silent=True. And I also get the following error when I try to re-run my set-up:

raise requests.HTTPError(rep.status_code, rep.url, message, response=rep)
requests.exceptions.HTTPError: [Errno 400] https://openalyx.internationalbrainlab.org/auth-token: 'Alyx authentication failed with credentials:

I need to access the IBL library for my research project. Is there any way for me to access to the public IBL database again? If so, how?

**Comment by mayofaulkner** [comment]
Hello,

Please could you run the following
```
from one.api import ONE
ONE.setup(base_url='https://openalyx.internationalbrainlab.org/', silent=True, make_default=True)
```
That should reset the connection and make sure you connect to the public IBL database.

Let me know if that doesn't fix the problem!

**Comment by bindeel** [comment]
Hi, thank you for getting back to me. Unfortunately, I'm still getting the following error even after running your code above:

raise requests.HTTPError(rep.status_code, rep.url, message, response=rep)
requests.exceptions.HTTPError: [Errno 400] https://openalyx.internationalbrainlab.org/auth-token: 'Alyx authentication failed with credentials:

**Comment by mayofaulkner** [comment]
Could you try running 
```
from one.api import ONE
ONE.setup(base_url='https://openalyx.internationalbrainlab.org/', silent=True, make_default=True)
one.alyx.authenticate(username='intbrainlab', password='international', force=True)
```

**Comment by bindeel** [comment]
Hi, I tried to run the code above and other iterations listed on the website to re-run my set-up but in every instance, I am asked to 

Enter Alyx password for "/Users/path to iblenv / bin/ python  /Users/path to IBL_database/IB"

Since I don't have an Alyx account/ password, I continue to get the same error above. 

I have tried to uninstall and reinstall iblenv and re-run the setups again or your suggested codes, but no luck still.

**Comment by k1o0** [comment]
Hi, please copy/paste the following code into your python console:
```
from one.api import ONE
ONE.setup(base_url='https://openalyx.internationalbrainlab.org/', silent=True, make_default=True)
one = ONE(username='intbrainlab', password='international', silent=False)
one.alyx.authenticate(password='international', force=True)
```
It should print `Connected to https://openalyx.internationalbrainlab.org/ as user "intbrainlab"`. After this, you should be able to instantiate ONE, the usual way: `from one.api import ONE; one = ONE()`.

Note that `silent=True` will mean it will prompt you for the password whenever needed. Only use `silent=False` when running scripts.  The next release of ONE (v2.10.0) will iron out a few of these issues you're having.

**Comment by bindeel** [comment]
Thank you for your help. 

When running the code above, it indeed printed `Connected to https://openalyx.internationalbrainlab.org/ as user "intbrainlab"`.

However, afterward, when I try to instantiate ONE the usual way after (using `from one.api import ONE; one = ONE()` ), It still prompts me to enter an Alyx password.

But, it seems like I can access and search within the public IBL as intended if I always use run commands after the following portion of your code above:
```
from one.api import ONE
ONE.setup(base_url='https://openalyx.internationalbrainlab.org/', silent=True, make_default=True)
one = ONE(username='intbrainlab', password='international', silent=False)
```

I still don't know if this poses an issue for future use, but I thought it might be important to let you know.

**Comment by k1o0** [comment]
Hi,

It is usual for ONE to prompt you for a password from time to time. The setup _does not_ store your password and it is unnecessary to run the setup function every time you use ONE. It is used to configure which database to use by default and the data repository information (both server and client-side).  If you do not explicitly pass in a password when you instantiate ONE, it will first check if you have a previously fetched auth token for the database and if not will prompt you for the password. When silent=True, it will instead just raise an error. The auth token can be stored for a long time but every so often expires and so you will need to re-authenticate. You can do this simply by instantiating without silent=True and then typing the password, or by passing in the password as an argument.  For the internal database we advise you to never use the password argument as it's very easy to accidentally publish your code with the password included.

In short, you do not need to call the setup method more than once.  I will close this issue for now but please feel free to reach out if you have more questions or problems.
